aptos-move/aptos-intent/src/codegen.rs

```diff
@@ -0,0 +1,489 @@
+        TypeParameter(idx) => TypeParameter(*subst_mapping.get(idx).unwrap()),
+    }
+}
+
+impl Context {
+    fn import_address(
+        &mut self,
+        module: &CompiledModule,
+        idx: AddressIdentifierIndex,
+    ) -> PartialVMResult<AddressIdentifierIndex> {
+        self.import_address_by_name(module.address_identifier_at(idx))
+    }
+
+    fn import_address_by_name(
+        &mut self,
+        address: &AccountAddress,
+    ) -> PartialVMResult<AddressIdentifierIndex> {
+        if let Some(result) = self.address_pool.get(address) {
+            return Ok(*result);
+        }
+        if self.script.address_identifiers.len() >= TableIndex::MAX as usize {
+            return Err(PartialVMError::new(StatusCode::INDEX_OUT_OF_BOUNDS));
+        }
+        let return_idx = AddressIdentifierIndex(self.script.address_identifiers.len() as u16);
+        self.script.address_identifiers.push(*address);
+        self.address_pool.insert(*address, return_idx);
+        Ok(return_idx)
+    }
+
+    fn import_identifier_by_name(&mut self, ident: &IdentStr) -> PartialVMResult<IdentifierIndex> {
+        if let Some(result) = self.identifier_pool.get(ident) {
+            return Ok(*result);
+        }
+        if self.script.identifiers.len() >= TableIndex::MAX as usize {
+            return Err(PartialVMError::new(StatusCode::INDEX_OUT_OF_BOUNDS));
+        }
+        let return_idx = IdentifierIndex(self.script.identifiers.len() as u16);
+        self.script.identifiers.push(ident.to_owned());
+        self.identifier_pool.insert(ident.to_owned(), return_idx);
+        Ok(return_idx)
+    }
+
+    fn import_identifier(
+        &mut self,
+        module: &CompiledModule,
+        idx: IdentifierIndex,
+    ) -> PartialVMResult<IdentifierIndex> {
+        self.import_identifier_by_name(module.identifier_at(idx))
+    }
+
+    fn import_module_by_id(&mut self, module: &ModuleId) -> PartialVMResult<ModuleHandleIndex> {
+        let address = self.import_address_by_name(module.address())?;
+        let name = self.import_identifier_by_name(module.name())?;
+        self.import_module_impl(address, name)
+    }
+
+    fn import_module(
+        &mut self,
+        module: &CompiledModule,
+        idx: ModuleHandleIndex,
+    ) -> PartialVMResult<ModuleHandleIndex> {
+        let handle = module.module_handle_at(idx);
+        let address = self.import_address(module, handle.address)?;
+        let name = self.import_identifier(module, handle.name)?;
+        self.import_module_impl(address, name)
+    }
+
+    fn import_module_impl(
+        &mut self,
+        address: AddressIdentifierIndex,
+        name: IdentifierIndex,
+    ) -> PartialVMResult<ModuleHandleIndex> {
+        if let Some(result) = self.module_pool.get(&(address, name)) {
+            return Ok(*result);
+        }
+        if self.script.module_handles.len() >= TableIndex::MAX as usize {
+            return Err(PartialVMError::new(StatusCode::INDEX_OUT_OF_BOUNDS));
```

Comment:
I know we have a history of abusing vm errors in tools, but wonder if we would benefit from this not repeating the same pattern.

Reply:
I think it would be fine this case since all errors will be turned into anyhow error here? Although I agree that we could remove those deps. WDYT?